### PR TITLE
Add support for Continuous Integration Builds in Drone

### PR DESCRIPTION
**Pull request by sureshc**
Migrate this repository's Continuous Integration service from CircleCI to self-hosted Drone: https://codedotorg.atlassian.net/browse/INF-95

### CODE REVIEW

File: .drone.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
+---
+kind: pipeline
+name: default
+
+steps:
+- name: restore-cache
```
Comment (uponthesun) [inline .drone.yml]: It seems like caching saves about a minute for the build? I think it'd be fine to not have caching to simplify the config file, thoughts?

Comment (sureshc) [reply .drone.yml]: Agreed.  I haven’t figured out how to cache `node_modules` based on the hash of `package.json`, which is what our CircleCI build currently does for this project.  I think we could just eliminate the caching.

**Comment by sureshc** [comment]
@uponthesun Can you show me how to get our drone server to sign this file?  Do I install Drone command line tools and the configure them to point to our drone server before running the signing tool?

**Comment by sureshc** [comment]
@joshlory I had to configure our codecov token to get the upload of the coverage report working (codecov does not require this for public repositories being built within CircleCI, but in Drone the token is required).  I’m not clear why the report is not being published to the GitHub Pull Request.  I could use a moment of your time to diagnose that.

**Comment by uponthesun** [comment]
@sureshc Yea, that's right. You can find the configuration values for the CLI by clicking on your user > user settings on the top right on the drone site.

**Pull request merged**